### PR TITLE
Comment hooks with special calls, fix extra chain

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | I wrote a script and checked all hook calls through the codebase that have some special parameters. Commented them all out. Removed extra parameters in some hook calls. Fixed an extra chaining I added by accident - there is a reference.
| Type?             | refacto
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | No QA needed if @PrestaShop/qa-functional approves.
| Fixed issue or discussion?     | 
| Related PRs       | 
| Sponsor company   | 

Auto test: https://github.com/Hlavtox/ga.tests.ui.pr/actions/runs/6156339419

### CODE REVIEW

**Comment by nicosomb** [comment]
@Hlavtox you have a conflict

**Comment by Hlavtox** [comment]
@nicosomb Fixed ;-)

File: classes/Hook.php [hunk 1]
```diff
@@ -807,10 +807,12 @@ public static function exec(
             Tools::displayParameterAsDeprecated('use_push');
         }
 
+        // If we in installation phase OR the hook is disabled, it won't be executed
```
Comment (LouisAUTHIE) [inline classes/Hook.php]: If we **are**

File: classes/Hook.php [hunk 2]
```diff
@@ -807,10 +807,12 @@ public static function exec(
...
         if (defined('PS_INSTALLATION_IN_PROGRESS') || !self::getHookStatusByName($hook_name)) {
             return $array_return ? [] : null;
         }
 
+        // Get hook register to collect debug information
```
Comment (LouisAUTHIE) [inline classes/Hook.php]: Hook **registry**

**Review by LouisAUTHIE — CHANGES_REQUESTED** [review]

File: classes/Hook.php (hunk 1 above)
Comment (Hlavtox) [reply classes/Hook.php]: Fixed :-)

File: classes/Hook.php (hunk 2 above)
Comment (Hlavtox) [reply classes/Hook.php]: Fixed :-)

**Comment by Hlavtox** [comment]
@LouisAUTHIE Can you recheck please? :-)

**Review by LouisAUTHIE — APPROVED** [review]

**Review by LouisAUTHIE — APPROVED** [review]

**Review by matks — APPROVED** [review]
Thank you it will be useful for module developers 👍

**Comment by matks** [comment]
@prestonBot I'm afraid there's a little problem 😅

**Comment by matks** [comment]
@Hlavtox Would you please run the UI Tests?

https://devdocs.prestashop-project.org/8/contribute/contribution-guidelines/ui-tests/

**Comment by Hlavtox** [comment]
@matks Done

**Comment by matks** [comment]
CI is broken we have lost MySQL
Capture d’écran 2023-09-12 à 09 50 51

**Comment by Hlavtox** [comment]
@matks All green! :-)

**Review by matks — APPROVED** [review]
This PR does not modify any behavior 👍

**Comment by matks** [comment]
As the [UI Tests](https://github.com/Hlavtox/ga.tests.ui.pr/actions/runs/6156339419) are green and no behavior has been modified I think we can merge it

**Comment by matks** [comment]
@Hlavtox Argh I just realized that was for 8.1.x and not develop although it is not a bugfix. Why submit this PR for patch version 😞 ?

**Comment by Hlavtox** [comment]
@matks It is a bugfix, I added extra chain that should not be there as it has no effect. ;-)

**Comment by Hlavtox** [comment]
And comments can go to 8.1.x, when somebody will dig in the current version, make it easier for him.

**Comment by matks** [comment]
> @matks It is a bugfix, I added extra chain that should not be there as it has no effect. ;-)

You did not solve a bug 😛 . You solved improveable code. It's different 😆

**Pull request merged**